aea/helpers/async_utils.py

```diff
@@ -63,37 +64,54 @@ def ensure_list(value: Any) -> List:
 class AsyncState:
     """Awaitable state."""
 
-    def __init__(self, initial_state: Any = None):
+    def __init__(
+        self, initial_state: Any = None, states_enum: Optional[Container[Any]] = None
+    ):
         """Init async state.
 
         :param initial_state: state to set on start.
+        :param states_enum: container of valid states if not provided state not checked on set.
         """
         self._state = initial_state
         self._watchers: Set[Future] = set()
-
-    @property
-    def state(self) -> Any:
-        """Return current state."""
-        return self.get()
-
-    @state.setter
-    def state(self, state: Any) -> None:
-        """Set state."""
-        self.set(state)
+        self._callbacks: List[Callable] = []
```

Comment:
Callables can be further typed as they are expected to consume single arg `state`

Reply:
yes, and also by return value, if it is relevant in this context.

Reply:
done